lib/telemetry_metrics_cloudwatch.ex

```diff
@@ -201,4 +222,14 @@ defmodule TelemetryMetricsCloudwatch do
 
     :ok
   end
+
+  @spec sample_measurement?(number()) :: boolean()
+  defp sample_measurement?(1), do: true
+  defp sample_measurement?(1.0), do: true
+  defp sample_measurement?(0), do: false
+  defp sample_measurement?(0.0), do: false
+
+  defp sample_measurement?(sample_rate) do
+    :rand.uniform() < sample_rate
+  end
```

Comment:
I think this'll work too (note I changed the comparison to be `<=` to the `sample_rate` too, which I think is right).

```suggestion
  defp sample_measurement?(sample_rate) when sample_rate == 1, do: true
  defp sample_measurement?(sample_rate) when sample_rate == 0, do: false
  defp sample_measurement?(sample_rate), do: :rand.uniform() <= sample_rate
```